ratchet_core/src/framed/mod.rs

```diff
@@ -488,6 +495,15 @@ where
         }
     }
 
+    pub async fn with_writer<'s, F, A>(&'s mut self, f: F) -> A::Output
```

Comment:
This is used in a different file (and only) there so should probably be moved over and made private rather than being public here. (Also, given the type contraints are longer than the function, I'd be tempted to just inline it).

Reply:
@horned-sphere I put it in this file as I don't want to make the members public so if I move it to where it is called from then I'll have to change the visibility